lib/remote_node.go

```diff
@@ -151,13 +156,24 @@ func NewHandshakeMetadata() *HandshakeMetadata {
 	return &HandshakeMetadata{}
 }
 
-func NewRemoteNode(id RemoteNodeId, validatorPublicKey *bls.PublicKey, isPersistent bool, srv *Server,
-	cmgr *ConnectionManager, keystore *BLSKeystore, params *DeSoParams, minTxFeeRateNanosPerKB uint64,
-	latestBlockHeight uint64, nodeServices ServiceFlag) *RemoteNode {
+func NewRemoteNode(
+	id RemoteNodeId,
+	validatorPublicKey *bls.PublicKey,
+	isPersistent bool,
+	isOutbound bool,
```

Comment:
At the time out creation, we need to define the expected `isOutbound` property for the remote node. That way, even if the connection fails to dial, we know what type of connection it was intended to be during cleanup